src/main/java/teammates/ui/controller/Action.java

```diff
@@ -106,10 +109,32 @@ protected void initialiseAttributes(HttpServletRequest req) {
         requestParameters = request.getParameterMap();
         session = request.getSession();
         sessionToken = CryptoHelper.computeSessionToken(session.getId());
+        parseAndInitializeRegkeyFromRequest();
         // Set error status forwarded from the previous action
         isError = getRequestParamAsBoolean(Const.ParamsNames.ERROR);
     }
 
+    /**
+     * Parses and initializes the regkey from the http request.
+     *
+     */
+    private void parseAndInitializeRegkeyFromRequest() {
+        String regkeyFromRequest = getRegkeyFromRequest();
+        if (regkeyFromRequest != null && regkeyFromRequest.contains("${amp}" + Const.ParamsNames.NEXT_URL + "=")) {
```

Comment:
Extract the condition `regkeyFromRequest != null && regkeyFromRequest.contains("${amp}" + Const.ParamsNames.NEXT_URL + "=")` to a self-explanatory variable?